scripts/config.py

```diff
@@ -159,46 +159,58 @@ def realfull_adapter(_name, active, section):
         return active
     return True
 
+# The goal of the full configuration is to have everything that can be tested
+# together. This includes deprecated or insecure options. It excludes:
```

Comment:
Note: we should make sure that everything that's excluded here is included in at least one component in `all.sh` - with possible exceptions for things that require special hardware or complex additional build dependencies. (For example, I think it's acceptable to keep not testing `PKCS11_C` until we remove it - after all, one of the reasons we're removing it is it has gone untested for ages).

I'm not suggesting to do it in this PR, just some time.

Reply:
As a data point, at this time, the following symbols from this list do not appear in `all.sh`:

* `MBEDTLS_REMOVE_3DES_CIPHERSUITES` (not tested as such, but doesn't gate any code as such)
* `MBEDTLS_SHA512_NO_SHA384` (tested via `depends-hashes.pl`)
* ` MBEDTLS_REMOVE_ARC4_CIPHERSUITES` (on by default)
* `MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT`
* `MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER` (meant to be tested via the PSA SPM crypto service)
* `MBEDTLS_PKCS11_C`
* `MBEDTLS_PSA_CRYPTO_SPM` (meant to be tested via the PSA SPM crypto service)
* `MBEDTLS_X509_ALLOW_UNSUPPORTED_CRITICAL_EXTENSION`
* `MBEDTLS_PSA_INJECT_ENTROPY`
* `MBEDTLS_HAVE_SSE2`

I agree that this PR is not the place to handle them.